### PR TITLE
DBAAS-135 change connection properties exposed to align with SBO mongo work

### PR DESCRIPTION
**Pull request by jianrongzhang89**
This PR is to parse the connection strings and store the derived host in the connection configmap.
Tested on the lab successfully.

### CODE REVIEW

File: bundle/manifests/mongodb-atlas-kubernetes.clusterserviceversion.yaml [hunk 1]
```diff
@@ -84,9 +86,6 @@ metadata:
             "credentialsRef": {
               "name": "my-atlas-key",
               "namespace": "mongodb-atlas-system"
-            },
-            "provider": {
-              "name": "mongodbatlas"
```
Comment (sgahlot) [inline bundle/manifests/mongodb-atlas-kubernetes.clusterserviceversion.yaml:89]: Do we not need this anymore?

File: pkg/controller/atlasconnection/atlasconnection_controller.go [hunk 2]
```diff
@@ -352,6 +353,18 @@ func (r *MongoDBAtlasConnectionReconciler) deleteDBUserFromAtlas(conn *dbaas.Mon
...
+	tokens := strings.Split(connectionStringStandardSrv, "//")
+	var host string
+	if len(tokens) < 2 {
+		host = connectionStringStandardSrv
+	} else {
+		host = tokens[1]
+	}
+	return host
+}
+
```
Comment (sgahlot) [inline pkg/controller/atlasconnection/atlasconnection_controller.go:367]: Looks good to me

**Review by redhatHameed — APPROVED** [review]

**Review by sgahlot — APPROVED** [review]
Looks good to me. I've added a comment - nothing major

File: bundle/manifests/mongodb-atlas-kubernetes.clusterserviceversion.yaml (hunk 1 above)
Comment (jianrongzhang89) [reply bundle/manifests/mongodb-atlas-kubernetes.clusterserviceversion.yaml:89]: That's right. It has been removed from the dbaas data structure.

File: pkg/controller/atlasconnection/atlasconnection_controller.go (hunk 2 above)
Comment (jianrongzhang89) [reply pkg/controller/atlasconnection/atlasconnection_controller.go:367]: thanks

**Review by jeremyary — COMMENTED** [review]
lgtm. Thanks!

**Pull request merged**